galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/routes/AuthTokensRoute.java

```diff
@@ -76,36 +76,76 @@ public AuthTokensRoute(
     /**
      * GET requests to /auth/tokens return all the tokens stored in the tokens
      * database, sorted by creation date order by default.
+     * This endpoint takes an optional query paramater 'loginId' for e.g loginId=admin
+     * Passing it returns a filtered list of token records stored in the auth store that matches the given login ID
      */
     @Override
     public HttpServletResponse handleGetRequest(String pathInfo, QueryParameters queryParams,
             HttpServletRequest request, HttpServletResponse response)
             throws ServletException, IOException, FrameworkException {
 
         logger.info("handleGetRequest() entered");
+        
+        List<AuthToken> tokensToReturn = new ArrayList<>();
+        String loginId = queryParams.getSingleString(AuthenticationServlet.QUERY_PARAM_LOGIN_ID, null);
+        
+        if(queryParams.checkAtLeastOneQueryParameterPresent("loginId")){
```

Comment:
Bit of inconsistency here - the login ID query parameter is specific to this route, so it would be good to move the QUERY_PARAM_LOGIN_ID constant in here.

Then in the following line, we're hard-coding the query parameter instead of using the constant.

Also, is `checkAtLeastOneQueryParameterPresent` the right function to use? We can just use the `validateLoginId` method since that's already checking if the login ID exists - this should be something like:

```java
String loginId = queryParams.getSingleString(QUERY_PARAM_LOGIN_ID, null);

if (loginId != null) {
    validateLoginId(loginId);
    tokensToReturn = getTokensByLoginId(loginId);
} else {
    tokensToReturn = getAllTokens();
}
```